Adds IPython magic to run in Jupyter notebooks

thomasjpfan opened the pull request:
Closes https://github.com/mityax/rustimport/issues/22

See this notebook for usage: https://gist.github.com/thomasjpfan/6997465eea97c6eff8cb7786e21ea1b4

- The IPython magic contains the `--release` and `--force` flag that does the same as the command line `python -m rustimport ...`
- I added a `--module-path-variable` to write the module path to a variable. This is useful for the Polars use case as show in the notebook. (I'm happy to remove this feature, if this is too much scope).

For the polars use case, the compiled module is written to a local `polars_lib` variable:

```
%%rustimport --module-path-variable=polars_lib
```

Later this can be used by polars to register it's plugin:

```python
import polars as pl
from polars.type_aliases import IntoExpr

@pl.api.register_expr_namespace("language")
class Language:
    ...

    def pig_latinnify(self) -> pl.Expr:
        return self._expr.register_plugin(
            lib=polars_lib,
        )
```

mityax commented:
This is amazing work, thank you!

The one thing I'd like to change is the extra dependency on ipython, since non-ipython users don't need it. I could imagine to a) hide it behind an "extra", i.e. ipython users would have to install "rustimport[ipython]" instead of "rustimport". The other option in my mind would be to b) conditionally import `ipython_magic`, only if we're running in ipython, and leave `ipython`  out of the `requirements.txt` and `setup.py` – I assume this should work since in an ipython environment we probably can rely on the `ipython` package being available anyways (?)

thomasjpfan commented:
> b) conditionally import ipython_magic, only if we're running in ipython, and leave ipython out of the requirements.txt and setup.py

I think making `ipython_magic` a conditional import is good enough. Anyone that wants to use the `%%rustimport` magic should already have IPython installed (either by installing Jupyter or IPython directly).

I updated this PR to remove IPython as a hard dependency and moved it to be a developer dependency.

mityax commented:
Thanks :)
